### ISSUE TITLE
Column count should depend on screen width, rather than orientation

### ISSUE THREAD

**Issue by stanch** [etsy/AndroidStaggeredGrid#148]
I have read #9 and #32. IMHO, width being greater or less than height (i.e. landscape or portrait respectively) has nothing to do with the optimum number of columns, since the grid is bounded horizontally, but not vertically. Instead, this value should be based solely on width, e.g.:

```
[0; 400dp) ⇒ 1 column,
[400dp; 610dp) ⇒ 2 columns,
[610dp; ∞) ⇒ 3 columns
```

For a more concrete example, I find that my grid looks best:
- with `1` column on a _Nexus 4_ portrait (`384 dp`),
- with `2` columns on a _Nexus 7_ portrait (`600 dp`), and
- with `3` columns in both _Nexus 4_ and _Nexus 7_ landscape (`640 dp` and `961 dp` respectively)

In XML the corresponding setting could look like this:

``` xml
app:column_count="0|400|610"
```

And in code:

``` java
grid.setColumnCount(new int[]{0, 400, 610});
```

Here are the values if you don’t want single columns:

``` xml
app:column_count="0|0|610"
```

``` java
grid.setColumnCount(new int[]{0, 0, 610});
```

---

If you don’t like this proposal, in theory the same effect could be achieved “manually”, using only the current `setColumnCount` API.
However, I would like to note that it seems broken. The orientation changes are handled correctly when using `setColumnCountPortrait` or `setColumnCountLandscape`, but using `setColumnCount` leads to #70, #105, #134, #138.

**Comment by maudem-airg** [comment]
Indeed, I just add this in my Fragment and it has been working like a charm! ^^
```
@Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        if(newConfig.orientation==Configuration.ORIENTATION_PORTRAIT) {
            mGridView.setColumnCountPortrait(getResources().getInteger(R.integer.grid_posts_column_count));
        }else if(newConfig.orientation==Configuration.ORIENTATION_LANDSCAPE){
            mGridView.setColumnCountLandscape(getResources().getInteger(R.integer.grid_posts_column_count));
        }
} 
```
